CONTRIBUTING.md

```diff
@@ -260,7 +260,15 @@ To do this you can:
 
 - Lastly, in the terminal run `terraform init` again and this time terraform will pull provider version from `tf_cache` network mirror. You can confirm this by noting the `Terraform has been successfully initialized! Using mongodb/mongodbatlas Vx.x.x from the shared cache directory` message.  
 
-
+## Code and Test Best Practices
+
+- Each resource (and associated data sources) is in a package in `internal/service`.
+- There can be multiple helper classes and they can also be used from other resources, e.g. `common_advanced_cluster.go` defines functions that are also used from other resources using `advancedcluster.FunctionName`.
+- Unit and Acceptances tests are in the same `_test.go` file. They are not in the same package as the code tests, e.g. `advancedcluster` tests are in `advancedcluster_test` package so coupling is minimized.
```

Comment:
Format for brevity consider adding chapters:

```suggestion

## Code Changes
...


## Unit and Acceptance tests
```


Contributing is usually focused on guiding new person to do specific tasks. 

Instead of

> - Unit and Acceptances tests are in the same `_test.go` file. They are not in the same package as the code tests, e.g. `advancedcluster` tests are in `advancedcluster_test` package so coupling is minimized.
Consider:

> Create *_test.go file for both Unit and Acceptance tests.
For example see: link.

> package so coupling is minimized.

Consider reader empathy - do we need that info for the new contributions?

Reply:
Those aren't blocking PR. Just some nits and CONTRIBUTING.md format tips for future.